pkg/dataloaders/middleware.go

```diff
@@ -0,0 +1,50 @@
+package dataloaders
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/graph-gophers/dataloader"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// Loaders wrap your data loaders to inject via middleware
+type Loaders struct {
+	UserInfoLoader *dataloader.Loader
+}
+
+type ctxKey string
+
+const (
+	loadersKey = ctxKey("dataloaders")
+)
+
+// NewLoaders instantiates data loaders for the middleware
+func NewLoaders(fetchUserInfos func(context.Context, []string) ([]*models.UserInfo, error)) *Loaders {
+	userInfoLoader := UserInfoLoader{
+		FetchUserInfos: fetchUserInfos,
+	}
+	loaders := &Loaders{
+		UserInfoLoader: dataloader.NewBatchedLoader(userInfoLoader.BatchUserInfos),
+	}
+	return loaders
+}
+
+// For returns the dataloader for a given context
+func For(ctx context.Context) *Loaders {
```

Comment:
it might be clearer to name this something more specific, like `CTXTWithLoaders`